src/surface/gui/gui/app.py

```
@@ -27,13 +29,22 @@ def run_gui(self) -> None:
         # Kills with Control + C
         signal.signal(signal.SIGINT, signal.SIG_DFL)
 
-        if self.theme_param.get_parameter_value().string_value == "dark":
-            qdarktheme.setup_theme()
-        elif self.theme_param.get_parameter_value().string_value == "watermelon":
-            # UGLY But WORKS
-            self.app.setStyleSheet("QWidget { background-color: green; color: pink; }")
+        # Apply theme
+        theme_param = self.node.get_parameter("theme").get_parameter_value().string_value
+        theme_path = os.path.join(get_package_share_directory("gui"),
+                                  "themes", theme_param + ".qss")
+
+        if theme_param == "dark":
+            with open(theme_path) as theme_file:
+                qdarktheme.setup_theme("dark", additional_qss="\n" + theme_file.read())
+
+        elif os.path.exists(theme_path):
+            with open(theme_path) as theme_file:
+                qdarktheme.setup_theme("light", additional_qss="\n" + theme_file.read())
+
         else:
             qdarktheme.setup_theme("light")
```

hmmm code duplication

this alternative is bad so maybe it's fine
```python
base_theme = "dark" if theme_param == "dark" else "light"
custom_styles = "\n"
if os.path.exists(theme_path):
    with open(theme_path) as theme_file:
        custom_styles += theme_file.read()

qdarktheme.setup_theme(base_theme, additional_qss=custom_styles)
```

---

The default style is kinda jank but I do like this better; updated

---

Isn't the pythonic way to try catch around this?

---

You could but that's just more lines in this case and `except: pass` is dumb